examples/dreamcast/cd/sound-dma/sound-dma.cpp

```diff
@@ -0,0 +1,67 @@
+#include <stdio.h>
+
+#include <unistd.h>
+#include <sys/ioctl.h>
+
+#include <dc/cdrom.h>
+#include <arch/cache.h>
+
+#define AICA_REG(n) (*(volatile uint32_t*)(0xA0700000+n))
+#define GD_PROTECTION  (*(volatile uint32_t*)0xA05F74B8)
+#define SYSCALLS_32(n) (*(volatile uint32_t*)(0xAC000000 + n))
+
+int main(int argc, char *argv[]) {
+
+   // Patch system calls to unprotect the GD-ROM DMA for the entire address space
+   SYSCALLS_32(0x1C20) = 0x8843007F;
+   SYSCALLS_32(0x23FC) = 0x8843007F;
+
+   // Flush the instruction cache for the patch
+   icache_flush_range(0xC000000, 0xC010000);
+
+   // Get the file sector and size
+   int fd = open("/cd/test.pcm", O_RDONLY);
```

Comment:
Do we need to check if the file could not be opened here? Like if someone doesn't know what they're doing and doesn't have the CD in the drive, so they get a decent error and know to insert the disc?

Reply:
done